runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -789,21 +863,21 @@ TR_J9JITServerSharedCache::TR_J9JITServerSharedCache(TR_J9VMBase *fe)
    }
 
 void *
-TR_J9JITServerSharedCache::pointerFromOffsetInSharedCache(void *offset)
+TR_J9JITServerSharedCache::pointerFromOffsetInSharedCache(uintptr_t offset)
    {
    TR_ASSERT(_stream, "stream must be initialized by now");
    // compute pointer from client's cache start address
    auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(_stream);
    return (void *) ((UDATA)offset + vmInfo->_cacheStartAddress);
    }
 
-void *
+uintptr_t
 TR_J9JITServerSharedCache::offsetInSharedCacheFromPointer(void *ptr)
    {
    TR_ASSERT(_stream, "stream must be initialized by now");
    // return offset from client's cache start address
    auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(_stream);
-   return (void *) ((UDATA)ptr - vmInfo->_cacheStartAddress);
+   return (uintptr_t)((UDATA)ptr - vmInfo->_cacheStartAddress);
```

Comment:
You have `_cacheStartAddress` in multiple places in this PR but this comment applies to all of them. This is not going to work in a multi-layered SCC environment. In a multi- env there are multiple "cache start addresses", not just one, so any reference to a single "cache start address" means that the code will probably have to be updated.

In this case I think we will have to send an array of cache start addresses, an array of cache sizes, and the number of caches to the server so that code like this can be updated.